src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypeDiagnosticAnalyzer.cs

```diff
@@ -109,16 +109,20 @@ protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, Seman
                 var variable = variableDeclaration.Variables.Single();
                 var initializer = variable.Initializer.Value;
 
-                // Discourage using "var pointer = stackalloc", after introducing C# 7.2, as it might be confusing to users expecting a Span<>
-                // Check descendant nodes as well, because Span-creating stackallocs can be nested in other nodes (like conditional operators and casts).
-                // https://github.com/dotnet/roslyn/issues/22768
-                if (initializer.DescendantNodesAndSelf().Any(node => node.IsKind(SyntaxKind.StackAllocArrayCreationExpression)))
+                // Do not suggest var replacement for stackalloc span expressions.
+                // This will change the bound type from a span to a pointer.
+                if (!variableDeclaration.Type.IsKind(SyntaxKind.PointerType)
+                    && initializer
+                        .DescendantNodesAndSelf(descendIntoChildren: node => !node.IsAnyLambdaOrAnonymousMethod())
+                        .Any(node => node.IsKind(SyntaxKind.StackAllocArrayCreationExpression)))
```

Comment:
FWIW, i would prefer really long expressions like this get broken out into either a helper method, or into statements.  i.e.:

```C#
var isPointerType = ...
var containsStackAlloc = ...

if (!isPointerType && containsStackAlloc) { ...
```

Reply:
It's just much more readable to me that way (IMO).